src/controller/controller.ts

```diff
@@ -499,8 +499,29 @@ export class Controller extends events.EventEmitter<ControllerEventMap> {
     private async changeChannel(oldChannel: number, newChannel: number): Promise<void> {
         logger.warning(`Changing channel from '${oldChannel}' to '${newChannel}'`, NS);
 
+        // According to the Zigbee specification:
+        // When broadcasting a Mgmt_NWK_Update_req to notify devices of a new channel, the nwkUpdateId parameter should be incremented in the NIB and included in the Mgmt_NWK_Update_req.
+        // The valid range of nwkUpdateId is 0x00 to 0xFF, and it should wrap back to 0 if necessary.
+        let nwkUpdateId: number = 0x00;
+        const isSupportsBackup = await this.adapter.supportsBackup();
+        if (isSupportsBackup) {
+            const backup = await this.adapter.backup(this.getDeviceIeeeAddresses());
```

Comment:
I don't think making a new backup is necessary here? Can't we get the value from the existing backup?

Reply:
Agreed.
Probably the property should be added to the adapter type NetworkParameters, since it's part of that, and then can be used throughout controller without issue (cached on start).
Remains to see if all adapters provide it though.

Reply:
Okay, I will try to change it to the implementation method you suggested.